bridge/jira/export.go

```diff
@@ -0,0 +1,449 @@
+				if !ok {
+					// Since an edit has to come after the creation, we expect we would
+					// have cached the creation id.
+					panic("unexpected error: comment id not found")
+				}
+				comment, err := client.UpdateComment(bugJiraID, commentID, opr.Message)
+				if err != nil {
+					err := errors.Wrap(err, "editing comment")
+					out <- core.NewExportError(err, b.Id())
+					return
+				}
+				out <- core.NewExportCommentEdition(op.Id())
+				// JIRA doesn't track all comment edits, they will only tell us about
+				// the most recent one. We must invent a consistent id for the operation
+				// so we use the comment ID plus the timestamp of the update, as
+				// reported by JIRA. Note that this must be consistent with the importer
+				// during ensureComment()
+				id = fmt.Sprintf("%s-%d", comment.ID, comment.Updated.Unix())
+			}
+
+		case *bug.SetStatusOperation:
+			jiraStatus, hasStatus := statusMap[opr.Status.String()]
+			if hasStatus {
+				exportTime, err = UpdateIssueStatus(client, bugJiraID, jiraStatus)
+				if err != nil {
+					err := errors.Wrap(err, "editing status")
+					out <- core.NewExportError(err, b.Id())
+					// Failure to update status isn't necessarily a big error. It's
+					// possible that we just don't have enough information to make that
+					// update. In this case, just don't export the operation.
+					continue
+				}
+				out <- core.NewExportStatusChange(op.Id())
+				// TODO(josh)[c2c6767]: query changelog to get the changelog-id so that
+				// we don't re-import the same change.
+				id = bugJiraID
+			} else {
+				out <- core.NewExportNothing(
+					op.Id(), fmt.Sprintf(
+						"No jira status mapped for %.8s", opr.Status.String()))
+			}
+
+		case *bug.SetTitleOperation:
+			exportTime, err = client.UpdateIssueTitle(bugJiraID, opr.Title)
+			if err != nil {
+				err := errors.Wrap(err, "editing title")
+				out <- core.NewExportError(err, b.Id())
+				return
+			}
+			out <- core.NewExportTitleEdition(op.Id())
+			// TODO(josh)[c2c6767]: query changelog to get the changelog-id so that
+			// we don't re-import the same change.
+			id = bugJiraID
+
+		case *bug.LabelChangeOperation:
+			exportTime, err = client.UpdateLabels(
+				bugJiraID, opr.Added, opr.Removed)
+			if err != nil {
+				err := errors.Wrap(err, "updating labels")
+				out <- core.NewExportError(err, b.Id())
+				return
+			}
+			out <- core.NewExportLabelChange(op.Id())
+			// TODO(josh)[c2c6767]: query changelog to get the changelog-id so that
+			// we don't re-import the same change.
+			id = bugJiraID
+
+		default:
+			panic("unhandled operation type case")
+		}
+
+		// mark operation as exported
+		// TODO(josh)[c2c6767]: Should we query the changelog after we export?
+		// Some of the operations above don't record an ID... so we are bound to
+		// re-import them. It shouldn't cause too much of an issue but we will have
+		// duplicate edit entries for everything and it would be nice to avoid that.
```

Comment:
especially because it should be possible to run the bridge in a cron job or something to have the update. If the data increase each time for nothing ...